### ISSUE TITLE
CORS only working when status code is 200

### ISSUE THREAD

**Issue by BurakKartal1** [fruitcake/laravel-cors#433]
**Before you start**
[x] Update to the latest version by running `composer update fruitcake/laravel-cors`

**Check your config**
[x]  Double-check your config file with the version from the repo. Make sure the `paths` property is correctly set. Start by allowing as much as possible.
[x]  Make sure the middleware is added to the global middleware in your Http Kernel (not group)

**Clear your caches**
Please do these steps again before submitting an issue:
[x]  Clear your config cache `php artisan config:clear`, route cache (`php artisan route:clear`) and normal cache (`php artisan cache:clear`).
[x]  Make sure your permissions are setup correctly (eg. storage is writable)

As title says i get same response when i have error on my Laravel App. Im using React for my Frontend. 

POST http://localhost/******** 500 (Internal Server Error)
Access to XMLHttpRequest at 'http://localhost/******' from origin 'http://localhost:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

**Comment by barryvdh** [comment]
Server error? Do you have anything in the logs? Does your application error? (eg because of CSRF token)

**Comment by BurakKartal1** [comment]
Its not about the error, its about CORS not working when anything "bad" happens on server side. I would like to see my Server Side Application's Error Message on my Console.log when i use React. i use Axios on my React if that affects in any way, which i dont think so. Because like i said i get CORS error which blocks anything that my Laravel is saying to me what went wrong. I can see my error message with my Postman. Because no CORS error ofc. 

I hope i was clear enough. There were same issues in this repo as this one. But when i searched them they always ended in irrelevant ways. 

#432  #426

I understand that you want to help, but you seem to be focused on only what the Error is. But problem is about CORS not working when error happens. Im not that experienced in laravel but i think its about  this package is not working on
````
/Exceptions/Handler.php
return parent::render($request, $exception);
````

i only added this package because it was on Upgrade guide --> https://laravel.com/docs/7.x/upgrade#cors-support

if its not essential and this bug is not going to be fixed or its not a bug in any technical way. I would like to use my own custom cors middleware.

Thanks.

**Comment by dloused** [comment]
same here...nuxt/vue frontend with laravel backend

**Comment by davidlomidze** [comment]
Same problem after updating to v1.0

**Comment by alkhachatryan** [comment]
> Server error? Do you have anything in the logs? Does your application error? (eg because of CSRF token)

CSRF token for RestfulAPI?

**Comment by stwinnie** [comment]
same problem here

**Comment by barryvdh** [comment]
Please create a new issue with more information.

**Comment by jamieshepherd** [comment]
Also having the same issue, seems to be a dupe of https://github.com/fruitcake/laravel-cors/issues/426 - Fresh laravel project, laravel-cors - any non-200 route (via api routes) from browser will return CORS errors instead of status codes.

**Comment by BurakKartal1** [comment]
First time i mentioned this issue was where i had forgotten a static function in my model that i forgot to implement. Thats why i had 500. But i also had other errors with 500 status code, those were fine. There was no CORS error. But again today i had this issue where my AppServiceProvider had simple "undefined variable" error. Instead of error message in my console there was CORS error. Which i had to use Postman to get what my Laravel application said.

I think its about how laravel processes its files. When something wrong happens in core files it doesnt even get to our CORS middleware ? Those can be fixed via server settings (#426) i guess.. but also that makes CORS middleware is a little bit useless. Am i missing something ?

**Comment by barryvdh** [comment]
Was this fixed?

**Comment by joke2k** [comment]
Hi everybody
I experienced the same behaviour described by @BurakKartal1 using react and having a server error returned with a response without CORS headers.

It was hard to debug but I think that I've found the reason, and is not about this package.

TL;DR
If you have a [Fatal error](https://www.php.net/manual/en/errorfunc.constants.php) (eg. assign a dynamic value to a const), laravel goes to shutdown,
skipping all the pipeline with middlewares (no cors headers), building a new response with the [`error_get_last()`](https://www.php.net/manual/en/function.error-get-last.php) data.

This package, correctly, adds the header also when an exception/error occours handling the [`Events\RequestHandled`](https://github.com/fruitcake/laravel-cors/blob/master/src/HandleCors.php#L49) event raised at the end of [`Kernel::handle()`](https://github.com/laravel/framework/blob/476ce4585321d6b02ab23e0bcf5c2353ae392bd6/src/Illuminate/Foundation/Http/Kernel.php#L116), but some kind of errors (fatals?) will be catched using [`register_shutdown_function`](https://www.php.net/manual/en/function.register-shutdown-function) and [`error_get_last()`](https://www.php.net/manual/en/function.error-get-last.php) as you can see in [`Foundation/Bootstrap/HandleExceptions`](https://github.com/laravel/framework/blob/476ce4585321d6b02ab23e0bcf5c2353ae392bd6/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php#L47)

I don't know how we can fix it, but I found this cute function [`trigger_error`](https://www.php.net/manual/en/function.trigger-error.php) that could helps to write a test.

At the end I can conclude that if I write bad php code, my React app will response without right CORS headers. In order to debug my errors I've to call the api directly and I'll the real error.
